src/mgr/PyModuleRegistry.cc

```diff
@@ -46,14 +47,56 @@ void PyModuleRegistry::init()
 
   // Set up global python interpreter
 #define WCHAR(s) L ## #s
+#if PY_VERSION_HEX >= 0x03080000
+  PyConfig py_config;
+  PyConfig_InitIsolatedConfig(&py_config);
+  BOOST_SCOPE_EXIT_ALL(&py_config) {
+    PyConfig_Clear(&py_config);
+  };
+  // site_import is 1 by default, but set it explicitly as we do import
+  // site packages manually for Python < 3.8
+  py_config.site_import = 1;
+
+  PyStatus status;
+  status = PyConfig_SetString(&py_config, &py_config.program_name, WCHAR(MGR_PYTHON_EXECUTABLE));
+  ceph_assert(PyStatus_Exception(status));
+  // Some python modules do not cope with an unpopulated argv, so lets
+  // fake one.  This step also picks up site-packages into sys.path.
+  const wchar_t* argv[] = {L"ceph-mgr"};
+  PyConfig_SetArgv(&py_config, 1, (wchar_t *const *)argv);
+  // Add more modules
+  if (g_conf().get_val<bool>("daemonize")) {
+    PyImport_AppendInittab("ceph_logger", PyModule::init_ceph_logger);
+  }
+  PyImport_AppendInittab("ceph_module", PyModule::init_ceph_module);
+  // Configure sys.path to include mgr_module_path
+  const auto mgr_module_path = g_conf().get_val<std::string>("mgr_module_path");
+  std::wstring module_search_paths(std::begin(mgr_module_path), std::end(mgr_module_path));
```

Comment:
should we assert if mgr_module_path isn't there?

Reply:
isn't where? in `py_config.module_search_paths`?

Reply:
if the configuration option 'mgr_module_path' is not set

Reply:
i see. but the default value this option is not empty, see https://github.com/ceph/ceph/blob/6847404eda0fe207e6991561eaf3df91aa0c8edf/src/common/options/mgr.yaml.in#L111 . if we consider it a bug, and really want to handle this case, i'd suggest create tracker ticket. and to implement it in another PR. as it is an preexisting behavior, and not in the scope of this change, which is created to address the deprecated Python APIs used in ceph-mgr. 

what do you think?

Reply:
I agree it's not a part of this PR.